File: Server/src/main/content/global/skill/production/crafting/armour/SnakeSkin.java
package content.global.skill.production.crafting.armour;

import core.game.node.item.Item;

/**
 * The enum Snake skin.
 */
public enum SnakeSkin {
    /**
     * The Snakeskin boot.
     */
    SNAKESKIN_BOOT(new Item(6328), 45, 30, 6),
    /**
     * The Snakeskin vambraces.
     */
    SNAKESKIN_VAMBRACES(new Item(6330), 47, 35, 8),
    /**
     * The Snakeskin bandana.
     */
    SNAKESKIN_BANDANA(new Item(6326), 48, 45, 5),
    /**
     * The Snakeskin chaps.
     */
    SNAKESKIN_CHAPS(new Item(6324), 51, 50, 12),
    /**
     * The Snakeskin body.
     */
    SNAKESKIN_BODY(new Item(6322), 53, 55, 15);

    private final Item product;
    private final int level;
    private final double experience;
    private final int requiredAmount;

    SnakeSkin(Item product, int level, double experience, int requiredAmount) {
        this.product = product;
        this.level = level;
        this.experience = experience;
        this.requiredAmount = requiredAmount;
    }

    /**
     * Gets product.
     *
     * @return the product
     */
    public Item getProduct() {
        return product;
    }

    /**
     * Gets level.
     *
     * @return the level
     */
    public int getLevel() {
        return level;
    }

    /**
     * Gets experience.
     *
     * @return the experience
     */
    public double getExperience() {
        return experience;
    }

    /**
     * Gets required amount.
     *
     * @return the required amount
     */
    public int getRequiredAmount() {
        return requiredAmount;
    }

}

File: Emulator/src/main/content/global/skill/production/crafting/data/Snakeskin.kt
package content.global.skill.production.crafting.data

import org.rs.consts.Items
import core.game.node.item.Item

/**
 * Represents the Snakeskin data.
 */
enum class Snakeskin(
    val product: Item,
    val level: Int,
    val experience: Double,
    val amount: Int
) {
    SNAKESKIN_BOOT(
        product = Item(Items.SNAKESKIN_BOOTS_6328),
        level = 45,
        experience = 30.0,
        amount = 6
    ),
    SNAKESKIN_VAMBRACES(
        product = Item(Items.SNAKESKIN_VBRACE_6330),
        level = 47,
        experience = 35.0,
        amount = 8
    ),
    SNAKESKIN_BANDANA(
        product = Item(Items.SNAKESKIN_BANDANA_6326),
        level = 48,
        experience = 45.0,
        amount = 5
    ),
    SNAKESKIN_CHAPS(
        product = Item(Items.SNAKESKIN_CHAPS_6324),
        level = 51,
        experience = 50.0,
        amount = 12
    ),
    SNAKESKIN_BODY(
        product = Item(Items.SNAKESKIN_BODY_6322),
        level = 53,
        experience = 55.0,
        amount = 15
    )
}